error_reporting.ts

```diff
@@ -0,0 +1,45 @@
+
+export class CompileTimeError extends Error {
+    __proto__: Error
+    constructor(message?: string) {
+     const trueProto = new.target.prototype;
+     super(message);
+ 
+     // Alternatively use Object.setPrototypeOf if you have an ES6 environment.
+     this.__proto__ = trueProto;
+   } 
+ }
+
+// I ❤️ TypeScript: https://github.com/microsoft/TypeScript/issues/13965
```

Comment:
Hahahahaha I have totally run into this before.

Reply:
Haha, yes we moved the implementation of "TypeCheckError" interface from the "type-check.ts" of the starter code to "error_reporting.ts". The comment got moved as well